client/app/bundles/system/admin/admin/components/SystemGetHelpFilter.tsx

```diff
@@ -0,0 +1,345 @@
+          disablePortal
+          getOptionLabel={(option): string => option.title}
+          onChange={(_, value): void => {
+            setSelectedFilter({
+              ...selectedFilter,
+              course: value,
+            });
+          }}
+          options={courseOptions}
+          renderInput={(params): JSX.Element => (
+            <TextField {...params} label={t(translations.filterCourseLabel)} />
+          )}
+          value={selectedFilter.course}
+        />
+      </Grid>
+      <Grid item paddingBottom={1} paddingRight={1} xs={1}>
+        <Autocomplete
+          clearOnEscape
+          disablePortal
+          getOptionLabel={(option): string => option.name}
+          onChange={(_, value): void => {
+            setSelectedFilter({
+              ...selectedFilter,
+              user: value,
+            });
+          }}
+          options={userOptions}
+          renderInput={(params): JSX.Element => (
+            <TextField {...params} label={t(translations.filterStudentLabel)} />
+          )}
+          value={selectedFilter.user}
+        />
+      </Grid>
+      <Grid item paddingBottom={1} paddingRight={1} xs={1}>
+        <TextField
+          fullWidth
+          InputLabelProps={{ shrink: true }}
+          label={t(translations.filterStartDateLabel)}
+          onChange={(e) =>
+            setSelectedFilter({ ...selectedFilter, startDate: e.target.value })
+          }
+          type="date"
+          value={selectedFilter.startDate || ''}
+        />
+      </Grid>
+      <Grid item paddingBottom={1} xs={1}>
+        <TextField
+          error={!!getDateValidationError(selectedFilter)}
+          fullWidth
+          InputLabelProps={{ shrink: true }}
+          label={t(translations.filterEndDateLabel)}
+          onChange={(e) =>
+            setSelectedFilter({ ...selectedFilter, endDate: e.target.value })
+          }
+          type="date"
+          value={selectedFilter.endDate || ''}
+        />
+      </Grid>
+    </Grid>
+  );
+};
+
+const FilterButtons: FC<
+  Pick<
+    Props,
+    'handleApplyFilter' | 'handleClearFilter' | 'setSelectedFilter'
+  > & {
+    selectedFilter: GetHelpFilter;
+  }
+> = ({
+  selectedFilter,
+  setSelectedFilter,
+  handleApplyFilter,
+  handleClearFilter,
+}) => {
+  const { t } = useTranslation();
+  const disableButton = Object.values(selectedFilter).every((x) => !x);
+
+  return (
+    <Grid item>
+      <Button
+        disabled={disableButton}
+        onClick={handleApplyFilter}
+        variant="contained"
+      >
+        {t(translations.applyFilterButton)}
+      </Button>
+      <Button
+        className="ml-10"
+        color="secondary"
+        disabled={disableButton}
+        onClick={() => {
+          setSelectedFilter({
+            course: null,
+            user: null,
+            startDate: '',
+            endDate: '',
+          });
+          handleClearFilter();
+        }}
+        variant="contained"
+      >
```

Comment:
For discussion: Personally I find the apply filter and clear filter buttons unintuitive. Why not trigger the query on field change?